crates/shared/src/http_solver/model.rs

```diff
@@ -185,19 +185,33 @@ impl Interaction for InteractionData {
     }
 }
 
+#[serde_as]
 #[derive(Copy, Clone, Debug, Deserialize, PartialEq, Serialize)]
-#[serde(rename_all = "camelCase")]
+#[serde(untagged)]
 pub enum Score {
-    /// The score used for ranking.
-    #[serde(with = "u256_decimal")]
-    Score(U256),
+    /// The score value is provided as is from solver.
+    /// Success probability is not incorporated into this value.
+    Solver {
+        #[serde(with = "u256_decimal")]
+        score: U256,
+    },
     /// This option is used to indicate that the solver did not provide a score.
     /// Instead, the score should be computed by the protocol.
     /// To have more flexibility, the protocol score can be tweaked by the
     /// solver by providing a discount.
-    #[serde(with = "u256_decimal")]
-    #[serde(rename = "scoreDiscount")]
-    Discount(U256),
+    Discount {
+        #[serde(with = "u256_decimal")]
+        #[serde(rename = "scoreDiscount")]
+        score_discount: U256,
+    },
+    /// This option is used to indicate that the solver did not provide a score.
+    /// Instead, the score should be computed by the protocol given the success
+    /// probability and optionally the amount of gas this settlement will take.
+    RiskAdjusted {
+        success_probability: f64,
+        #[serde_as(as = "Option<DecimalU256>")]
+        gas_amount: Option<U256>,
+    },
```

Comment:
For consistency, should this also be:

```suggestion
    #[serde(rename_all = "camelCase")]
    RiskAdjusted {
        success_probability: f64,
        #[serde_as(as = "Option<DecimalU256>")]
        gas_amount: Option<U256>,
    },
```

Reply:
It can, just seemed that ref_token and a few other fields use the _ notation, not sure what our policy on this json is (it honestly doesn't look like we have any consistency)

Reply:
Ah, I missed that. In that case it's very unfortunate that `scoreDiscount` is the only camel case field. I just noticed that the old type had `rename_all camelCase` - hence my comment.